assets/src/dashboard/components/typeahead/typeaheadOptions/index.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { forwardRef, useEffect, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { KEYS } from '../../../constants';
+import { DROPDOWN_ITEM_PROP_TYPE } from '../../types';
+import { Menu, MenuItem, MenuItemContent } from './components';
+
+const TypeaheadOptions = forwardRef(function TypeaheadOptions(
+  { handleFocusToInput, selectedIndex, items = [], onSelect },
+  listRef
+) {
+  const [hoveredIndex, setHoveredIndex] = useState(0);
+  const [focusedIndex, setFocusedIndex] = useState(0);
+
+  useEffect(() => {
+    const handleKeyDown = (event) => {
+      switch (event.key) {
+        case KEYS.UP:
+          event.preventDefault();
+          if (focusedIndex === 0) {
+            handleFocusToInput();
+          }
+          if (focusedIndex > 0 && listRef.current) {
+            listRef.current.children[focusedIndex - 1].focus();
+          }
+          break;
+
+        case KEYS.DOWN:
+          event.preventDefault();
+          if (focusedIndex < items.length - 1 && listRef.current) {
+            listRef.current.children[focusedIndex + 1].focus();
+          }
+          break;
+
+        case KEYS.ENTER:
+          event.preventDefault();
+          if (onSelect) {
+            onSelect(items[focusedIndex]);
+          }
+          break;
+
+        case KEYS.ESC:
+          event.preventDefault();
+          handleFocusToInput();
+          break;
```

Comment:
This updated a bit - now it's just to control the custom keyboard interactions. focus is set `onFocus` in list items